app/lib/activitypub/activity/create.rb

```diff
@@ -25,7 +25,7 @@ def perform
   private
 
   def find_existing_status
-    status   = Status.find_by(uri: object_uri)
+    status   = status_from_uri(object_uri)
     status ||= Status.find_by(uri: @object['_:atomUri']) if @object['_:atomUri'].present?
```

Comment:
Should this also be status_from_uri? Since atomUri could be local too, maybe..? 🤔